server/stream.go

```diff
@@ -3570,18 +3570,15 @@ func (mset *stream) processJetStreamMsg(subject, reply string, hdr, msg []byte,
 	}
 
 	if err == nil && seq > 0 && numConsumers > 0 {
-		mset.mu.Lock()
+		mset.mu.RLock()
 		for _, o := range mset.consumers {
 			o.mu.Lock()
-			if o.isLeader() {
-				if seq > o.lsgap && o.isFilteredMatch(subject) {
-					o.sgap++
-				}
+			if o.isLeader() && o.isFilteredMatch(subject) {
```

Comment:
We used to signal if o.isLeader() was true, but now will only if leader and isFilteredMatch(). Is that ok?

Reply:
We would signal consumers that might not match, which would not affect behavior, but would wake them up and they would go back into waitForMsgs().

Reply:
Gotcha.